walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -82,104 +101,81 @@
-    TimeRange[] endOrderedTimeRanges = getOrderedTimeRanges(occupiedTimeRanges,
-                                                                TimeRange.ORDER_BY_END);
+    List<TimePoint> timePoints = getTimePointsFromTimeRanges(occupiedTimeRanges);
+    List<TimePoint> orderedTimePoints = getOrderedTimePoints(timePoints);
     List<TimeRange> availableTimeRanges = new LinkedList<>();
 
     // Potentially add the {@code TimeRange} before the first occupied {@code TimeRange}.
-    TimeRange firstStartingTimeRange = startOrderedTimeRanges[0];
     checkDurationAndAddAvailableTimeRange(requestedDuration,
-                                          TimeRange.START_OF_DAY, firstStartingTimeRange.start(),
+                                          TimeRange.START_OF_DAY,
+                                          orderedTimePoints.get(0).getTime(),
                                           availableTimeRanges);
-    findAndAddGapsInOccupiedTimeRanges(requestedDuration,
-                                       startOrderedTimeRanges, endOrderedTimeRanges,
-                                       availableTimeRanges);
+    findAndAddGapsInOccupiedTimeRanges(requestedDuration, orderedTimePoints, availableTimeRanges);
     // Potentially add the {@code TimeRange} after the last occupied {@code TimeRange}.
-    TimeRange lastEndingTimeRange = endOrderedTimeRanges[endOrderedTimeRanges.length - 1];
     checkDurationAndAddAvailableTimeRange(requestedDuration,
-                                          lastEndingTimeRange.end(), TimeRange.END_OF_DAY + 1,
+                                          orderedTimePoints.get(orderedTimePoints.size() - 1)
+                                              .getTime(),
+                                          TimeRange.END_OF_DAY + 1,
                                           availableTimeRanges);
     return availableTimeRanges;
   }
 
   /**
-   * Sorts {@code timeRanges} based on the specified {@code sortOrder}.
+   * Finds the start and end {@code TimePoint} of {@code timeRanges}.
+   */
+  private List<TimePoint> getTimePointsFromTimeRanges(Collection<TimeRange> timeRanges) {
+    List<TimePoint> timePoints = new ArrayList<>(2 * timeRanges.size());
+    for (TimeRange timeRange : timeRanges) {
+      timePoints.add(new TimePoint(timeRange.start(), true));
+      timePoints.add(new TimePoint(timeRange.end(), false));
+    }
+    return timePoints;
+  }
+
+  /**
+   * Sorts the start or end{@code TimePoint} based on ascending chronological order.
    */
-  private TimeRange[] getOrderedTimeRanges(Collection<TimeRange> timeRanges,
-      Comparator<TimeRange> sortOrder) {
-    List<TimeRange> orderedTimeRanges = new LinkedList<>(timeRanges);
-    Collections.sort(orderedTimeRanges, sortOrder);
-    return orderedTimeRanges.toArray(new TimeRange[0]);
+  private List<TimePoint> getOrderedTimePoints(List<TimePoint> timePoints) {
+    Collections.sort(timePoints, TimePoint.ORDER_BY_TIME);
+    return timePoints;
   }
```

Comment:
Maybe combine these into a getSortedTimePointsFromTimeRanges()

Reply:
Sure! Thank you.